filters/apiusagemonitoring/spec.go

```diff
@@ -242,23 +246,13 @@ func (s *apiUsageMonitoringSpec) buildPathInfoListFromConfiguration(apis []*apiC
 			}
 			existingPathPattern[pathPattern] = info
 
-			// Compile the regular expression
-			var err error
-			var pathPatternMatcher *regexp.Regexp
-			pathPatternMatcherI, ok := regCache.Load(pathPattern)
-			if !ok {
-				pathPatternMatcher, err = regexp.Compile(pathPattern)
-				if err != nil {
-					regCache.Store(pathPattern, nil)
-					log.Errorf(
-						"args[%d].path_templates[%d] ignored: error compiling regular expression %q for path %q: %v",
-						apiIndex, templateIndex, pathPattern, info.PathTemplate, err)
-					continue
-				}
-				regCache.Store(pathPattern, pathPatternMatcher)
-
-			} else {
-				pathPatternMatcher = pathPatternMatcherI.(*regexp.Regexp)
+			pathPatternMatcher, err := loadOrCompileRegex(pathPattern)
+			if err != nil {
+				regCache.Store(pathPattern, nil)
```

Comment:
this regCache.Store is unnecessary, because loadOrCompileRegex stores the nil for the pathPattern

Reply:
I think you have an outdated view, not?

Reply:
thanks, true, fixed